composer.json

```diff
@@ -7,8 +7,8 @@
   "keywords": ["module", "xp"],
   "require" : {
     "xp-framework/core": "^12.0 | ^11.6 | ^10.16",
-    "xp-framework/reflection": "^3.0 | ^2.13",
-    "xp-framework/ast": "^11.1",
+    "xp-framework/reflection": "dev-feature/asymmetric-visibility as 3.2.0",
+    "xp-framework/ast": "dev-feature/asymmetric-visibility as 11.3.0",
     "php" : ">=7.4.0"
```

Comment:
Bumping the dependency on `xp-framework/reflection` to 3.2 is a BC break and would result in a major release of XP Compiler unless we backport this feature to 2.14

Reply:
Backporting this looked good until the following difference:

| Modifier         | PHP 7.0..7.3 | PHP 7.4+ | Comment               |
| ---------------- | ------------ | -------- | --------------------- |
| IS_STATIC        | 1            | 16       |                       |
| IS_ABSTRACT      | 2            | 64       |                       |
| IS_FINAL         | 4            | 32       |                       |
| IS_PUBLIC        | 256          | 1        |                       |
| IS_PROTECTED     | 512          | 2        |                       |
| IS_PRIVATE       | 1024         | 4        |                       |
| IS_READONLY      |              | 128      | (as of PHP 8.1)       |
| IS_PUBLIC_SET    |              | 1024     | (XP only)             |
| IS_PROTECTED_SET |              | 2048     |                       |
| IS_PRIVATE_SET   |              | 4096     |                       |

Our (explicit) IS_PUBLIC_SET clashes with IS_PRIVATE. We need to work around this by using an unused bit!

Reply:
Fixed in xp-framework/reflection@1053dcb and xp-framework/reflection@1bb8c25

Reply:
https://github.com/xp-framework/reflection/releases/tag/v2.15.0